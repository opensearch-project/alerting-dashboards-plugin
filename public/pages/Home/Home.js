/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { Component } from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';
import { EuiTab, EuiTabs } from '@elastic/eui';

import Dashboard from '../Dashboard/containers/Dashboard';
import Monitors from '../Monitors/containers/Monitors';
import DestinationsList from '../Destinations/containers/DestinationsList';

const getSelectedTabId = (pathname) => {
  if (pathname.includes('monitors')) return 'monitors';
  if (pathname.includes('destinations')) return 'destinations';
  return 'alerts';
};

export default class Home extends Component {
  constructor(props) {
    super(props);
    const {
      location: { pathname },
    } = this.props;
    const selectedTabId = getSelectedTabId(pathname);
    this.state = { selectedTabId };
    this.tabs = [
      {
        id: 'alerts',
        name: 'Alerts',
        route: 'dashboard',
      },
      {
        id: 'monitors',
        name: 'Monitors',
        route: 'monitors',
      },
      {
        id: 'destinations',
        name: 'Destinations',
        route: 'destinations',
      },
    ];
  }

  componentDidUpdate(prevProps) {
    const {
      location: { pathname: prevPathname },
    } = prevProps;
    const {
      location: { pathname: currPathname },
    } = this.props;
    if (prevPathname !== currPathname) {
      const selectedTabId = getSelectedTabId(currPathname);
      // eslint-disable-next-line react/no-did-update-set-state
      this.setState({ selectedTabId });
    }
  }

  onSelectedTabChanged = (route) => {
    const {
      location: { pathname: currPathname },
    } = this.props;
    if (!currPathname.includes(route)) {
      this.props.history.push(route);
    }
  };

  renderTab = (tab) => (
    <EuiTab
      onClick={() => this.onSelectedTabChanged(tab.route)}
      isSelected={tab.id === this.state.selectedTabId}
      key={tab.id}
    >
      {tab.name}
    </EuiTab>
  );

  render() {
    const { httpClient, notifications, setFlyout, landingDataSourceId, defaultRoute } = this.props;
    return (
      <div>
        {!defaultRoute && (
          <EuiTabs size="s">
            {this.tabs.map(this.renderTab)}
          </EuiTabs>
        )}
        <div style={{ padding: '25px 25px' }}>
          <Switch>
            <Route
              exact
              path="/dashboard"
              render={(props) => (
                <Dashboard
                  {...props}
                  httpClient={httpClient}
                  notifications={notifications}
                  perAlertView={false}
                  setFlyout={setFlyout}
                  landingDataSourceId={landingDataSourceId}
                />
              )}
            />
            <Route
              exact
              path="/monitors"
              render={(props) => (
                <Monitors
                  {...props}
                  httpClient={httpClient}
                  notifications={notifications}
                  landingDataSourceId={landingDataSourceId}
                />
              )}
            />
            <Route
              exact
              path="/destinations"
              render={(props) => (
                <DestinationsList
                  {...props}
                  httpClient={httpClient}
                  notifications={notifications}
                />
              )}
            />
            <Redirect to={defaultRoute || "/dashboard"} />
          </Switch>
        </div>
      </div>
    );
  }
}
