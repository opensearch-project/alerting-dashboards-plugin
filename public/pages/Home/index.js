/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import Home from './Home';

export default Home;
