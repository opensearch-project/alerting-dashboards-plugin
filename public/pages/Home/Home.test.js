/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { HashRouter as Router, Route } from 'react-router-dom';
import { render } from 'enzyme';

import Home from './Home';

describe('Home', () => {
  test('renders', () => {
    const component = (
      <Router>
        <Route render={(props) => <Home httpClient={{}} {...props} />} />
      </Router>
    );

    expect(render(component)).toMatchSnapshot();
  });
});
