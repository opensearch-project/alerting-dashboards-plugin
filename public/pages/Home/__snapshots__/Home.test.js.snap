// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders 1`] = `
<div>
  <div
    class="euiTabs euiTabs--small"
    role="tablist"
    style="padding:16px 16px 0px"
  >
    <button
      aria-selected="true"
      class="euiTab euiTab-isSelected"
      role="tab"
      type="button"
    >
      <span
        class="euiTab__content"
      >
        Alerts
      </span>
    </button>
    <button
      aria-selected="false"
      class="euiTab"
      role="tab"
      type="button"
    >
      <span
        class="euiTab__content"
      >
        Monitors
      </span>
    </button>
    <button
      aria-selected="false"
      class="euiTab"
      role="tab"
      type="button"
    >
      <span
        class="euiTab__content"
      >
        Destinations
      </span>
    </button>
  </div>
  <div
    style="padding:16px"
  />
</div>
`;
